src/components/common/Accordion/index.tsx

```diff
@@ -2,13 +2,21 @@ import { Disclosure } from '@headlessui/react';
 import { FlexContainer, Text } from '@/components';
 import { ChevronUpIcon } from '@heroicons/react/20/solid';
 import { AccordionProps } from '@/interfaces';
+import { useState } from 'react';
 
 const Accordion = ({ title, children, open = false }: AccordionProps) => {
+  const [isSelected, setIsSelected] = useState(open);
+
   return (
     <Disclosure defaultOpen={open}>
       {({ open }) => (
         <>
-          <Disclosure.Button className='flex w-full justify-between items-center rounded border px-2 py-1 strong-text hover:bg-gray-200'>
+          <Disclosure.Button
+            onClick={() => setIsSelected(!isSelected)}
```

Comment:
don't overcomplicate it. We've done something similiar in Shiksha course page. take a look. We don't need all this